src/Engine/ProtoCore/Parser/atg/Associative.atg

```diff
@@ -1055,14 +1055,30 @@ Associative_DecoratedIdentifier<out ProtoCore.AST.AssociativeAST.AssociativeNode
                         .)
         ':'
 
-        ident
                         (. 
-                            int type = core.TypeSystem.GetType(t.val); 
+                            string ident = string.Empty;
+							int type = ProtoCore.DSASM.Constants.kInvalidIndex;
+							if (la.kind == 1)
+							{
+								if (IsIdentList())
+								{
+									Associative_IdentifierList(out node);
+									ident = node.ToString();
+								}
+								else
+								{
+						.)
+		ident
+						(.
+									ident = t.val;
+								}
+							}
+							type = core.TypeSystem.GetType(ident);
```

Comment:
Is there an instance where you parse and the type is not yet found? what will it return?